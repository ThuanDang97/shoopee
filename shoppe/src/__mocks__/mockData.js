export const PRODUCT_ITEM = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: '-25%',
}

export const PRODUCT_DETAIL = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: '-25%',
  quantity: 1,
  introduction:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. ',
}

export const LIST_PRODUCT = [
  {
    id: 1,
    imageUrl: '/images/liraEarring.webp',
    name: 'Lira Earrings',
    price: 20,
    status: '-25%',
  },
  {
    id: 2,
    imageUrl: '/images/liraEarring.webp',
    name: 'Lira Earrings',
    price: 20,
    status: '',
  },
  {
    id: 3,
    imageUrl: '/images/liraEarring.webp',
    name: 'Kaede Hair Pin Set Of 3',
    price: 20,
    status: '',
  },
  {
    id: 4,
    imageUrl: '/images/liraEarring.webp',
    name: 'Hair Pin Set of 3',
    price: 20,
  },
  {
    id: 5,
    imageUrl: '/images/liraEarring.webp',
    name: 'Plaine Necklace',
    price: 20,
    status: 'Sold out',
  },
  {
    id: 6,
    imageUrl: '/images/liraEarring.webp',
    name: 'Yuki Hair Pin Set of 3',
    price: 20,
    status: '',
  },
]

export const CARTS_MOCK = [
  {
    name: 'Lira Earrings',
    price: 20,
    introduction:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. ',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. Nunc sed porta augue. Morbi porta tempor odio, in molestie diam bibendum sed.',
    imageUrl:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/Img%2001.png?alt=media&token=7437b1da-7f3b-4ecb-9bbf-e0fca1df9d9b',
    status: '- %21',
    id: 1,
    quantity: 17,
  },
  {
    name: 'Hal Earrings',
    price: 25,
    introduction:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. ',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. Nunc sed porta augue. Morbi porta tempor odio, in molestie diam bibendum sed.',
    imageUrl:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/Img%2002.png?alt=media&token=598ddd11-aa7b-4e7f-ba58-97cf6d8cdb1a',
    status: '',
    id: 2,
    quantity: 28,
  },
]

export const MOCKED_PRODUCTS = [
  {
    id: 1,
    imageUrl: '/images/liraEarring.webp',
    name: 'Lira Earrings',
    price: 20,
    status: '-25%',
  },
  {
    id: 2,
    imageUrl: '/images/liraEarring.webp',
    name: 'Lira Earrings',
    price: 20,
    status: '',
  },
  {
    id: 3,
    imageUrl: '/images/liraEarring.webp',
    name: 'Kaede Hair Pin Set Of 3',
    price: 20,
    status: '',
  },
  {
    id: 4,
    imageUrl: '/images/liraEarring.webp',
    name: 'Hair Pin Set of 3',
    price: 20,
  },
  {
    id: 5,
    imageUrl: '/images/liraEarring.webp',
    name: 'Plaine Necklace',
    price: 20,
    status: 'Sold out',
  },
  {
    id: 6,
    imageUrl: '/images/liraEarring.webp',
    name: 'Yuki Hair Pin Set of 3',
    price: 20,
    status: '',
  },
]

export const MOCK_PRODUCT = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: '-25%',
  quantity: 1,
  introduction:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. ',
}
