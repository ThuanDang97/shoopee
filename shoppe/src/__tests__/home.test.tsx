import React from 'react'

// Pages
import Home, { getStaticProps } from '@pages/products'

// Utils
import { render } from '@utils/testUtils'

// Mocks
import { MOCKED_PRODUCTS } from '@mocks/mockData'

jest.mock('@services/index', () => {
  const { LIST_PRODUCT } = require('@mocks/mockData')

  return {
    fetcherApi: jest.fn().mockResolvedValue(LIST_PRODUCT),
  }
})

describe('Home render', () => {
  it('Should show match Home DOM Snapshot', async () => {
    const { container } = render(<Home products={MOCKED_PRODUCTS} error="" />)

    expect(container).toMatchSnapshot()
  })

  it('Should getStaticProps return data correctly ', async () => {
    const mockedStaticProps = {
      props: {
        products: MOCKED_PRODUCTS,
      },
    }

    const value = await getStaticProps({})
    expect(value).toEqual(mockedStaticProps)

    jest.resetAllMocks()
  })
})
