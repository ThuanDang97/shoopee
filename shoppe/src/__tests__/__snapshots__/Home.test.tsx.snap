// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render Should show match Home DOM Snapshot 1`] = `
<div>
  <div
    class="css-nmq16m"
  >
    <h2
      class="chakra-heading css-lpqg55"
    >
      Shop The Latest
    </h2>
    <div
      class="css-cy7va0"
    >
      <div
        class="css-1i5wex2"
      >
        <div
          class="css-pbwc9d"
        >
          <input
            class="chakra-input css-b2ycb2"
            placeholder="Search..."
            type="text"
            value=""
          />
          <button
            aria-label="Search database"
            class="chakra-button css-xcn56a"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-bokek7"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <h2
          class="chakra-heading css-cmaddm"
        >
          Shop
        </h2>
      </div>
      <div
        class="css-1t0pur5"
      >
        <div
          class="css-42arw8"
        >
          <div
            class="css-1xcdsqd"
          >
            <div
              class="css-ervdjc"
            >
              <p
                class="chakra-text css-1ok93o9"
              >
                -25%
              </p>
            </div>
            <img
              alt="Lira Earrings"
              data-nimg="1"
              decoding="async"
              height="300"
              src="/images/liraEarring.webp"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="300"
            />
            <div
              class="css-inp0u2"
            >
              <h3
                class="chakra-heading css-b14ccn"
              >
                <a
                  href="/products/1"
                >
                  Lira Earrings
                </a>
              </h3>
              <p
                class="chakra-text css-uc37v3"
              >
                $ 20.00
              </p>
            </div>
          </div>
          <div
            class="css-1xcdsqd"
          >
            <img
              alt="Lira Earrings"
              data-nimg="1"
              decoding="async"
              height="300"
              src="/images/liraEarring.webp"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="300"
            />
            <div
              class="css-inp0u2"
            >
              <h3
                class="chakra-heading css-b14ccn"
              >
                <a
                  href="/products/2"
                >
                  Lira Earrings
                </a>
              </h3>
              <p
                class="chakra-text css-uc37v3"
              >
                $ 20.00
              </p>
            </div>
          </div>
          <div
            class="css-1xcdsqd"
          >
            <img
              alt="Kaede Hair Pin Set Of 3"
              data-nimg="1"
              decoding="async"
              height="300"
              src="/images/liraEarring.webp"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="300"
            />
            <div
              class="css-inp0u2"
            >
              <h3
                class="chakra-heading css-b14ccn"
              >
                <a
                  href="/products/3"
                >
                  Kaede Hair Pin Set Of 3
                </a>
              </h3>
              <p
                class="chakra-text css-uc37v3"
              >
                $ 20.00
              </p>
            </div>
          </div>
          <div
            class="css-1xcdsqd"
          >
            <img
              alt="Hair Pin Set of 3"
              data-nimg="1"
              decoding="async"
              height="300"
              src="/images/liraEarring.webp"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="300"
            />
            <div
              class="css-inp0u2"
            >
              <h3
                class="chakra-heading css-b14ccn"
              >
                <a
                  href="/products/4"
                >
                  Hair Pin Set of 3
                </a>
              </h3>
              <p
                class="chakra-text css-uc37v3"
              >
                $ 20.00
              </p>
            </div>
          </div>
          <div
            class="css-1xcdsqd"
          >
            <div
              class="css-ervdjc"
            >
              <p
                class="chakra-text css-1ok93o9"
              >
                Sold out
              </p>
            </div>
            <img
              alt="Plaine Necklace"
              data-nimg="1"
              decoding="async"
              height="300"
              src="/images/liraEarring.webp"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="300"
            />
            <div
              class="css-inp0u2"
            >
              <h3
                class="chakra-heading css-b14ccn"
              >
                <a
                  href="/products/5"
                >
                  Plaine Necklace
                </a>
              </h3>
              <p
                class="chakra-text css-uc37v3"
              >
                $ 20.00
              </p>
            </div>
          </div>
          <div
            class="css-1xcdsqd"
          >
            <img
              alt="Yuki Hair Pin Set of 3"
              data-nimg="1"
              decoding="async"
              height="300"
              src="/images/liraEarring.webp"
              style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
              width="300"
            />
            <div
              class="css-inp0u2"
            >
              <h3
                class="chakra-heading css-b14ccn"
              >
                <a
                  href="/products/6"
                >
                  Yuki Hair Pin Set of 3
                </a>
              </h3>
              <p
                class="chakra-text css-uc37v3"
              >
                $ 20.00
              </p>
            </div>
          </div>
        </div>
        <button
          class="chakra-button css-5p8da"
          data-loading=""
          disabled=""
          type="button"
        >
          <div
            class="chakra-button__spinner chakra-button__spinner--start css-1aamklj"
          >
            <div
              class="chakra-spinner css-1wpg9jd"
            >
              <span
                class="css-8b45rq"
              >
                Loading...
              </span>
            </div>
          </div>
          <span
            class="css-q7lffx"
          >
            Load More
          </span>
        </button>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
